content/documents/implementing-single-sign-on-with-google-idp.mdx

```diff
@@ -0,0 +1,413 @@
+  HashKey: "supersecrethash",
+  ElevatedRoles: ["BuyerUserAdmin"],
+})
+.catch((err:OrderCloudError) => console.log(err));
+console.log(integrationEvent);`,
+csharp: `using OrderCloud.SDK;\n
+var client = new OrderCloudClient(...);\n
+await client.AuthenticateAsync();\n
+IntegrationEvent response = await client.IntegrationEvents.CreateAsync(new IntegrationEvent {
+    ID = "openidconnect",
+    Name = "openidconnect",
+    EventType = IntegrationEventType.OpenIDConnect,
+    CustomImplementationUrl = "{your-ngrok-url}/integration-events",
+    HashKey = "supersecrethash",
+    ElevatedRoles = new ApiRole[] { ApiRole.BuyerUserAdmin },
+});`}}
+/>
+
+| OrderCloud Property     | Description                                                                                                                                                                                                                                                   |
+|-------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| ID                      | Unique identifier for the integration event                                                                                                                                                                                                                   |
+| Name                    | A short name describing the integration event, this is not user facing                                                                                                                                                                                        |
+| EventType               | Indicates what type of integration event this is, in our case we should use OpenIDConnect                                                                                                                                                                     |
+| CustomImplementationUrl | This indicates the base URL of your middleware where OrderCloud should post to. For OpenIDConnect it will call out to the path /createuser and /syncuser                                                                                                      |
+| HashKey                 | This is an important security feature that is used by your middleware to validate that requests made to your endpoints are legitimate and come from OrderCloud                                                                                                |
+| ElevatedRoles           | An optional array of roles that will encoded in the user's token and sent along in the payload to /createuser and /syncuser. In our case we are defining BuyerUserAdmin so that our middleware endpoints have the roles necessary to create users on the fly. |
```

Comment:
An optional array of roles that will encode* in the user's token and send* along in the payload to /createuser and /syncuser.